pip/qsharp/telemetry.py

```diff
@@ -0,0 +1,305 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""
+This module sends telemetry directly to Azure Monitor using a similar mechanism and
+format to the Azure Monitor OpenTelemetry Python SDK. It only supports custom metrics of
+type "counter" and "histogram" for now. It's goal is to be minimal in size and dependencies,
+and easy to read to understand exactly what data is being sent.
+
+To use this API, simply call `log_telemetry` with the metric name, value, and any other
+optional properties. The telemetry will be batched and sent at a regular intervals (60 sec),
+and when the process is about to exit.
+
+Disable qsharp Python telemetry by setting the environment variable `QSHARP_PYTHON_TELEMETRY=none`.
+"""
+
+import atexit
+import json
+import locale
+import logging
+import os
+import platform
+import time
+import urllib.request
+import warnings
+
+from datetime import datetime, timezone
+from queue import SimpleQueue, Empty
+from threading import Thread
+from typing import Any, Dict, Literal, List, TypedDict, Union
+
+logger = logging.getLogger(__name__)
+
+QSHARP_VERSION = "0.0.0.dev0"
+
+AIKEY = os.environ.get("QSHARP_PYTHON_AI_KEY") or "5ef527ea-7943-4e42-866d-e3b291d44e2b"
+AIURL = (
+    os.environ.get("QSHARP_PYTHON_AI_URL")
+    or "https://westus2-2.in.applicationinsights.azure.com//v2.1/track"
+)
+
+# Environment variables take precedence, else disable telemetry for non 'stable' builds
+QSHARP_PYTHON_TELEMETRY = (os.environ.get("QSHARP_PYTHON_TELEMETRY") or "").lower()
+TELEMETRY_ENABLED = (
+    True
+    if QSHARP_PYTHON_TELEMETRY in ["1", "true", "enabled"]
+    else (
+        False
+        if QSHARP_PYTHON_TELEMETRY in ["0", "false", "disabled", "none"]
```

Comment:
Do you want this to be case sensitive?

Reply:
No. That's why I explicitly did it this way with the `lower()` on line 43.